src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1290,21 +1301,20 @@ private void updateToolFlag(@NotNull final String tool, final int required)
      */
     private int getMostEfficientTool(@NotNull final Block target)
     {
-        final String tool = target.getHarvestTool(target.getDefaultState());
+        final ToolType toolType = ToolType.getToolType(target.getHarvestTool(target.getDefaultState()));
         final int required = target.getHarvestLevel(target.getDefaultState());
         int bestSlot = -1;
         int bestLevel = Integer.MAX_VALUE;
         @NotNull final InventoryCitizen inventory = worker.getInventoryCitizen();
-        final int hutLevel = worker.getWorkBuilding().getBuildingLevel();
+        final int maxToolLevel = worker.getWorkBuilding().getMaxToolLevel();
 
         for (int i = 0; i < new InvWrapper(worker.getInventoryCitizen()).getSlots(); i++)
         {
             final ItemStack item = inventory.getStackInSlot(i);
-            final int level = Utils.getMiningLevel(item, tool);
-
+            final int level = Utils.getMiningLevel(item, toolType);
             if (level >= required && level < bestLevel)
             {
-                if (tool == null || InventoryUtils.verifyToolLevel(item, level, hutLevel))
+                if (toolType == ToolType.NONE || InventoryUtils.verifyToolLevel(item, level, required, maxToolLevel))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)